src/Control/Effect/Diagnostics.hs

```diff
@@ -65,6 +71,14 @@ fromEither = either fatal pure
 fromEitherShow :: (Show err, Has Diagnostics sig m) => Either err a -> m a
 fromEitherShow = either (fatal . T.pack . show) pure
 
+-- | Lift a Maybe result into Diagnostics, with the given diagnostic thrown for @Nothing@
+fromMaybe :: (ToDiagnostic err, Has Diagnostics sig m) => err -> Maybe a -> m a
+fromMaybe msg = maybe (fatal msg) pure
+
+-- | Lift a Maybe result into Diagnostics, with a Text error message for @Nothing@
+fromMaybeText :: Has Diagnostics sig m => Text -> Maybe a -> m a
+fromMaybeText = fromMaybe
```

Comment:
Why is this needed?

Reply:
normally, because of `OverloadedStrings`, `fromMaybe "foo"` will have an ambiguous `IsString` constraint. We can force it by using a type application `fromMaybe @Text "foo"` or we can add this version that's monomorphic for `Text`